tests/Microsoft.DotNet.Framework.Docker.Tests/ImageTests.cs

```diff
@@ -0,0 +1,100 @@
+    {
+        private const string WSC2016 = "windowsservercore-ltsc2016";
+        private const string WSC1709 = "windowsservercore-1709";
+
+        private static string OsFilter => Environment.GetEnvironmentVariable("IMAGE_OS_FILTER");
+        private static string VersionFilter => Environment.GetEnvironmentVariable("IMAGE_VERSION_FILTER");
+
+        private static ImageDescriptor[] TestData = new ImageDescriptor[]
+            {
+                new ImageDescriptor { RuntimeVersion = "3.5", BuildVersion = "3.5", OsVariant = WSC2016 },
+                new ImageDescriptor { RuntimeVersion = "3.5", BuildVersion = "3.5", OsVariant = WSC1709 },
+                new ImageDescriptor { RuntimeVersion = "4.6.2", BuildVersion = "4.7.1", OsVariant = WSC2016 },
+                new ImageDescriptor { RuntimeVersion = "4.7", BuildVersion = "4.7.1", OsVariant = WSC2016 },
+                new ImageDescriptor { RuntimeVersion = "4.7.1", BuildVersion = "4.7.1", OsVariant = WSC2016 },
+                new ImageDescriptor { RuntimeVersion = "4.7.1", BuildVersion = "4.7.1", OsVariant = WSC1709 },
+            };
+
+        private DockerHelper DockerHelper { get; set; }
+
+        public ImageTests(ITestOutputHelper outputHelper)
+        {
+            DockerHelper = new DockerHelper(outputHelper);
+        }
+
+        public static IEnumerable<object[]> GetVerifyImagesData()
+        {
+            string versionFilterPattern = VersionFilter != null ? GetFilterRegexPattern(VersionFilter) : null;
+            string osFilterPattern = OsFilter != null ? GetFilterRegexPattern(OsFilter) : null;
+
+            // Filter out test data that does not match the active os and version filters.
+            return TestData
+                .Where(imageDescriptor => OsFilter == null
+                    || Regex.IsMatch(imageDescriptor.OsVariant, osFilterPattern, RegexOptions.IgnoreCase))
+                .Where(imageDescriptor => VersionFilter == null
+                    || Regex.IsMatch(imageDescriptor.RuntimeVersion, versionFilterPattern, RegexOptions.IgnoreCase))
+                .Select(imageDescriptor => new object[] { imageDescriptor });
+        }
+
+        private static string GetFilterRegexPattern(string filter)
+        {
+            return $"^{Regex.Escape(filter).Replace(@"\*", ".*").Replace(@"\?", ".")}$";
+        }
+
+        [Theory]
+        [MemberData(nameof(GetVerifyImagesData))]
+        public void VerifyImages(ImageDescriptor imageDescriptor)
+        {
+            string baseBuildImage = $"microsoft/dotnet-framework-build:{imageDescriptor.BuildVersion}-{imageDescriptor.OsVariant}";
+            VerifyImageExist(baseBuildImage);
+
+            string baseRuntimeImage = $"microsoft/dotnet-framework:{imageDescriptor.RuntimeVersion}-{imageDescriptor.OsVariant}";
+            VerifyImageExist(baseRuntimeImage);
+
+            string appId = $"dotnetapp-{DateTime.Now.ToFileTime()}";
+            string workDir = Path.Combine(
```

Comment:
Maybe throw if the folder does not exist?

Reply:
The docker build will fail if the folder does not exist.  I don't value in pre-checking.  Leaving as is.